### ISSUE TITLE
Ctrl+C causes `bst shell` to exit when I don't want it to

### ISSUE THREAD

**Issue by cs-shadow** [cs-shadow/buildstream#44]
In GitLab by @samthursfield on Jun 21, 2017, 11:06

I find that whenever I hit Ctrl+C in a `bst shell`, it exits the shell. I pretty much never want Ctrl+C to exit the shell, I'm usually pressing this to cancel a running build. I will type `exit` or hit Ctrl+D when I want to actually exit the sandbox.

**Comment by cs-shadow** [comment]
In GitLab by @tristanvb on Jun 22, 2017, 11:11

closed via commit a7aa1e24c96ce56b7b5bc9e1b80a08e55bee11b2

**Comment by cs-shadow** [comment]
In GitLab by @tristanvb on Jun 22, 2017, 11:11

This is fixed with the following commit:

```
commit a7aa1e24c96ce56b7b5bc9e1b80a08e55bee11b2
Author: Tristan Van Berkom <tristan.vanberkom@codethink.co.uk>
Date:   Thu Jun 22 20:06:08 2017 +0900

    _sandboxbwrap.py: Fix job control in interactive mode
    
    For some reason, launching bwrap under a new session (setsid()) causes
    the shell launched in that container to lack job control.
    
    This patch fixes this by conditionally creating the new session, which
    we need to ensure we handle SIGTSTP and SIGTERM separately in regular
    non-interactive mode, but not creating the session when launched in
    interactive mode.
    
    This fixes issue #40
```
